diff --git a/config/compiler/BUILD.gn b/config/compiler/BUILD.gn
index 71d7b1236..24e0bcbac 100644
--- a/config/compiler/BUILD.gn
+++ b/config/compiler/BUILD.gn
@@ -2176,7 +2176,7 @@ config("default_warnings") {
       # TODO(thakis): Remove this once
       # https://swiftshader-review.googlesource.com/c/SwiftShader/+/57968 has
       # rolled into angle.
-      cflags += [ "/wd4244" ]
+      # cflags += [ "/wd4244" ]
     }
   } else {
     if (is_apple || is_android) {
@@ -2546,7 +2546,7 @@ config("no_chromium_code") {
     }
     cflags += [
       "/wd4800",  # Disable warning when forcing value to bool.
-      "/wd4267",  # TODO(jschuh): size_t to int.
+      # "/wd4267",  # TODO(jschuh): size_t to int.
     ]
   } else if (is_clang) {
     cflags += [ "-Wall" ]
diff --git a/config/win/BUILD.gn b/config/win/BUILD.gn
index eb25c10ac..6ac2feca8 100644
--- a/config/win/BUILD.gn
+++ b/config/win/BUILD.gn
@@ -515,16 +515,16 @@ config("default_crt") {
     # Component mode: dynamic CRT. Since the library is shared, it requires
     # exceptions or will give errors about things not matching, so keep
     # exceptions on.
-    configs = [ ":dynamic_crt" ]
+    configs = [ ":dynamic_crt", ":win_msvc_cfg" ]
   } else {
     if (current_os == "winuwp") {
       # https://blogs.msdn.microsoft.com/vcblog/2014/06/10/the-great-c-runtime-crt-refactoring/
       # contains a details explanation of what is happening with the Windows
       # CRT in Visual Studio releases related to Windows store applications.
-      configs = [ ":dynamic_crt" ]
+      configs = [ ":dynamic_crt", ":win_msvc_cfg" ]
     } else {
       # Desktop Windows: static CRT.
-      configs = [ ":static_crt" ]
+      configs = [ ":static_crt", ":win_msvc_cfg" ]
     }
   }
 }
@@ -702,3 +702,11 @@ config("nominmax") {
 config("wincrypt_use_symbol_prefix") {
   defines = [ "WINCRYPT_USE_SYMBOL_PREFIX" ]
 }
+
+# Control Flow Guard (CFG)
+config("win_msvc_cfg") {
+  if (!is_clang) {
+    cflags = [ "/guard:cf", "/Qspectre", "/W3" ]
+    ldflags = [ "/guard:cf" ]
+  }
+}
diff --git a/toolchain/win/setup_toolchain.py b/toolchain/win/setup_toolchain.py
index e0b7c03a5..65ba26248 100644
--- a/toolchain/win/setup_toolchain.py
+++ b/toolchain/win/setup_toolchain.py
@@ -190,6 +190,8 @@ def _LoadToolchainEnv(cpu, toolchain_root, sdk_dir, target_store):
     # building with a new and untested SDK. This should stay in sync with the
     # packaged toolchain in build/vs_toolchain.py.
     args.append(SDK_VERSION)
+    # Use Spectre runtime libraries
+    args.append('-vcvars_spectre_libs=spectre')
     variables = _LoadEnvFromBat(args)
   return _ExtractImportantEnvironment(variables)
 
