diff --git a/BUILD.gn b/BUILD.gn
index 382c7d3ed44..7ef6a37baaf 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -8117,34 +8117,6 @@ group("v8_python_base") {
   data = [ ".vpython3" ]
 }
 
-# Enable fallbacks for files that might be hard to access on Clusterfuzz.
-copy("v8_clusterfuzz_fallbacks") {
-  testonly = true
-  sources = [ "test/mjsunit/wasm/wasm-module-builder.js" ]
-  outputs = [ "$root_out_dir/test/mjsunit/wasm/wasm-module-builder.js" ]
-}
-
-group("v8_clusterfuzz") {
-  testonly = true
-
-  deps = [
-    ":d8",
-    ":v8_clusterfuzz_fallbacks",
-    ":v8_simple_inspector_fuzzer",
-    "tools/clusterfuzz/trials:v8_clusterfuzz_resources",
-  ]
-
-  if (v8_multi_arch_build) {
-    deps += [
-      ":d8(//build/toolchain/linux:clang_x64)",
-      ":d8(//build/toolchain/linux:clang_x64_v8_arm64)",
-      ":d8(//build/toolchain/linux:clang_x86)",
-      ":d8(//build/toolchain/linux:clang_x86_v8_arm)",
-      ":d8(tools/clusterfuzz/foozzie/toolchain:clang_x64_fuzzer_experiments)",
-    ]
-  }
-}
-
 # Targets we ensure work with gcc. The aim is to keep this list small to have
 # a fast overall compile time.
 group("v8_gcc_light") {
@@ -8407,7 +8379,7 @@ v8_executable("d8") {
   }
 
   if (v8_correctness_fuzzer) {
-    deps += [ "tools/clusterfuzz/foozzie:v8_correctness_fuzzer_resources" ]
+    # deps += [ "tools/clusterfuzz/foozzie:v8_correctness_fuzzer_resources" ]
   }
 
   defines = []
@@ -8847,3 +8819,9 @@ if (current_toolchain == default_toolchain) {
 
   write_file("$root_build_dir/gn_logs.txt", _lines)
 }
+
+group("jsi") {
+  deps = [
+    "jsi:v8jsi",
+  ]
+}
diff --git a/DEPS b/DEPS
index b2c1f763079..dd13feb6172 100644
--- a/DEPS
+++ b/DEPS
@@ -853,6 +853,17 @@ hooks = [
                Var('rbe_instance'),
                ],
   },
+  {
+    'name': 'rc_win',
+    'pattern': '.',
+    'condition': 'checkout_win and host_os == "win"',
+    'action': [ 'download_from_google_storage',
+                '--no_resume',
+                '--no_auth',
+                '--bucket', 'chromium-browser-clang/rc',
+                '-s', 'build/toolchain/win/rc/win/rc.exe.sha1',
+    ],
+  },
 ]
 
 recursedeps = [
diff --git a/src/heap/cppgc/marking-state.h b/src/heap/cppgc/marking-state.h
index 73bd45147bc..3694933f555 100644
--- a/src/heap/cppgc/marking-state.h
+++ b/src/heap/cppgc/marking-state.h
@@ -364,7 +364,7 @@ class MutatorMarkingState final : public BasicMarkingState {
   ~MutatorMarkingState() override = default;
 
   inline bool MarkNoPush(HeapObjectHeader& header) {
-    return MutatorMarkingState::BasicMarkingState::MarkNoPush(header);
+    return BasicMarkingState::MarkNoPush(header);
   }
 
   inline void ReTraceMarkedWeakContainer(cppgc::Visitor&, HeapObjectHeader&);
diff --git a/src/maglev/maglev-ir.h b/src/maglev/maglev-ir.h
index a94c1b89ab7..a05906621e3 100644
--- a/src/maglev/maglev-ir.h
+++ b/src/maglev/maglev-ir.h
@@ -2867,8 +2867,8 @@ class NodeTMixin : public BaseNode {
   template <typename... Args>
   explicit NodeTMixin(uint64_t bitfield, Args&&... args)
       : BaseNode(bitfield, std::forward<Args>(args)...) {
-    DCHECK_EQ(this->NodeBase::opcode(), NodeBase::opcode_of<Derived>);
-    DCHECK_EQ(this->NodeBase::properties(), Derived::kProperties);
+    //DCHECK_EQ(this->NodeBase::opcode(), NodeBase::opcode_of<Derived>);
+    //DCHECK_EQ(this->NodeBase::properties(), Derived::kProperties);
   }
 };
 
@@ -2941,7 +2941,7 @@ class FixedInputNodeTMixin : public NodeTMixin<BaseT, Derived> {
   template <typename... Args>
   explicit FixedInputNodeTMixin(uint64_t bitfield, Args&&... args)
       : NodeTMixin<BaseT, Derived>(bitfield, std::forward<Args>(args)...) {
-    DCHECK_EQ(this->NodeBase::input_count(), kInputCount);
+    //DCHECK_EQ(this->NodeBase::input_count(), kInputCount);
   }
 };
 
diff --git a/src/utils/allocation.cc b/src/utils/allocation.cc
index 9fe80b267d2..2b0890aaebd 100644
--- a/src/utils/allocation.cc
+++ b/src/utils/allocation.cc
@@ -67,8 +67,10 @@ const int kAllocationTries = 2;
 }  // namespace
 
 v8::PageAllocator* GetPlatformPageAllocator() {
-  DCHECK_NOT_NULL(GetPageAllocatorInitializer()->page_allocator());
-  return GetPageAllocatorInitializer()->page_allocator();
+  //DCHECK_NOT_NULL(GetPageAllocatorInitializer()->page_allocator());
+  //return GetPageAllocatorInitializer()->page_allocator();
+
+  return V8::GetCurrentPlatform()->GetPageAllocator();
 }
 
 v8::VirtualAddressSpace* GetPlatformVirtualAddressSpace() {
diff --git a/third_party/googletest/BUILD.gn b/third_party/googletest/BUILD.gn
index 1cf84b3d8f4..d5c5ec9521e 100644
--- a/third_party/googletest/BUILD.gn
+++ b/third_party/googletest/BUILD.gn
@@ -88,8 +88,8 @@ source_set("gtest") {
 
   all_dependent_configs = [ ":gtest_config" ]
 
-  configs -= [ "//build/config/compiler:chromium_code" ]
-  configs += [ "//build/config/compiler:no_chromium_code" ]
+  configs -= [ "//build/config/compiler:chromium_code", "//build/config/compiler:no_rtti", "//build/config/compiler:no_exceptions" ]
+  configs += [ "//build/config/compiler:no_chromium_code", "//build/config/compiler:rtti", "//build/config/compiler:exceptions" ]
 
   # V8-only workaround for http://crbug.com/chromium/1191946. Ensures that
   # googletest is compiled with the same visibility such as the rest of V8, see
diff --git a/tools/v8windbg/BUILD.gn b/tools/v8windbg/BUILD.gn
index f33de70b304..18a3e038d69 100644
--- a/tools/v8windbg/BUILD.gn
+++ b/tools/v8windbg/BUILD.gn
@@ -23,7 +23,7 @@ source_set("v8windbg_base") {
     "DbgEng.lib",
     "DbgModel.lib",
     "RuntimeObject.lib",
-    "comsuppwd.lib",
+    "comsuppw.lib",
   ]
 
   public_configs = [ ":v8windbg_config" ]
